internal/util/function/minhash_function.go

```diff
@@ -0,0 +1,664 @@
+
+	// check input field count
+	if len(funSchema.GetInputFieldNames()) != 1 {
+		return fmt.Errorf("minhash function should only have one input field, but now %d", len(funSchema.GetInputFieldNames()))
+	}
+	if len(funSchema.GetOutputFieldNames()) != 1 {
+		return fmt.Errorf("minhash function should only have one output field, but now %d", len(funSchema.GetOutputFieldNames()))
+	}
+
+	// Find fields by name (since FieldIDs may not be assigned yet during validation)
+	inputFieldName := funSchema.GetInputFieldNames()[0]
+	outputFieldName := funSchema.GetOutputFieldNames()[0]
+
+	for _, field := range collSchema.GetFields() {
+		if field.GetName() == inputFieldName {
+			inputField = field
+		}
+		if field.GetName() == outputFieldName {
+			outputField = field
+		}
+	}
+
+	if inputField == nil {
+		return fmt.Errorf("minhash function input field '%s' not found", inputFieldName)
+	}
+	if outputField == nil {
+		return fmt.Errorf("minhash function output field '%s' not found", outputFieldName)
+	}
+
+	if inputField.GetDataType() != schemapb.DataType_VarChar && inputField.GetDataType() != schemapb.DataType_String {
+		return fmt.Errorf("minhash function input field '%s' is not string type, is %s",
+			inputFieldName, inputField.GetDataType())
+	}
+	// check function params
+	numHashes := int(-1)
+	for _, param := range funSchema.GetParams() {
+		if strings.ToLower(param.GetKey()) == NumHashesKey {
+			val, err := strconv.ParseInt(param.GetValue(), 10, 64)
+			if err == nil {
+				numHashes = int(val)
+				if numHashes <= 0 {
+					return fmt.Errorf("Param num_hashes:%s must be positive", param.GetValue())
+				}
+				break
+			}
+		} else if strings.ToLower(param.GetKey()) == ShingleSizeKey {
+			val, err := strconv.ParseInt(param.GetValue(), 10, 64)
+			if err == nil {
+				shingleSize := int(val)
+				if shingleSize <= 0 {
+					return fmt.Errorf("Param shingle_size:%s must be positive", param.GetValue())
+				}
+			}
+		}
+	}
+	// check numHashes with output field
+	var outputDim int64 = -1
+	if outputField.GetDataType() != schemapb.DataType_BinaryVector {
+		return fmt.Errorf("minhash function output field '%s' is not binary vector type", outputFieldName)
+	}
+	for _, param := range outputField.GetTypeParams() {
+		if param.GetKey() == "dim" {
+			val, err := strconv.ParseInt(param.GetValue(), 10, 64)
+			if err == nil {
+				outputDim = val
+				break
+			}
+		}
+	}
+
+	if numHashes > 0 {
+		expectedDim := int64(numHashes * 32) // binary vector, each hash is 4 bytes (32 bits), but stored as 8 bits in binary vector
+		if outputDim != expectedDim {
+			return fmt.Errorf("minhash function output field '%s' dim %d does not match expected dim %d (numHashes %d * one minhash signature size of 32bit)", outputFieldName, outputDim, expectedDim, numHashes)
+		}
+	} else {
+		if outputDim%32 != 0 {
+			return fmt.Errorf("minhash function output field '%s' dim %d is not multiple of 32 (one minhash signature size)", outputFieldName, outputDim)
```

Comment:
Why is this constraint still needed? Isn’t this part already set to drop by default?